### ISSUE TITLE
include appSecretName for github in applicationSet (as in argocd 2.5)

### ISSUE THREAD

**Issue by Fettah** [argoproj/argo-cd#10928]
we need to keep to date to the main argocd project and include their latest updated to their custom resource definitions.
the applicationSet has some new fields like the "appSecretName" for github
https://github.com/argoproj/argo-cd/blob/v2.5.0-rc2/manifests/crds/applicationset-crd.yaml#L2375
would be great to see that included at some points

**Comment by crenshaw-dev** [comment]
@Fettah I'm not sure I understand the request. Did you mean to post this in https://github.com/argoproj/argo-helm ?

**Comment by jared-schmidt-niceincontact** [comment]
I agree with @Fettah.  The argocd app authentication has been added to applicationset generators, but there is no documentation or examples showing how it works or that it even exists.

**Comment by Fettah** [comment]
> @Fettah I'm not sure I understand the request. Did you mean to post this in https://github.com/argoproj/argo-helm ?

yes
